scripts/csv_importer.py

```diff
@@ -0,0 +1,273 @@
+        for name in usernames:
+            print(name)
+
+        username = None
+        while username not in usernames:
+            username = input("Collection owner name: ")
+
+    cursor.execute(f"SELECT id FROM koi_user WHERE username='{username}'")
+    owner_id = cursor.fetchone()[0]
+
+    if create_new_collection:
+        collection_id = insert_collection(owner_id, collection_name)
+        insert_log(owner_id, collection_id, collection_name, "App\\Entity\\Collection")
+    else:
+        while collection_name not in collection_names:
+            collection_name = input("Collection name: ")
+            print(collection_names, type(collection_names))
+
+        cursor.execute(f"SELECT id FROM koi_collection WHERE title='{collection_name}'")
+        collection_id = cursor.fetchone()[0]
+
+    for item in items:
+        item_id = insert_item(owner_id, collection_id, item[name_field])
+
+        for index, attribute_name in enumerate(headers):
+            if attribute_name != name_field:
+                insert_datum(owner_id, item_id, attribute_name, item[attribute_name], index + 1,
+                             "private" if attribute_name in private_fields else "public")
+
+        insert_log(owner_id, item_id, item[name_field], "App\\Entity\\Item")
+
+    cursor.execute(f"SELECT cached_values FROM koi_collection WHERE id='{collection_id}'")
+    cached_values = ast.literal_eval(cursor.fetchone()[0])
+    cached_values["counters"]["items"] += len(items)
+    cursor.execute(f"UPDATE koi_collection SET cached_values=%s, updated_at=%s WHERE id='{collection_id}'",
+                   (json.dumps(cached_values), get_current_time()))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Import data from a csv file to Koillection database")
+    parser.add_argument("-u", "--user", nargs='?', help="Database user", type=str)
+    parser.add_argument("-p", "--password", nargs='?', help="Database password", type=str)
+    parser.add_argument("--host", nargs='?', help="Database host", type=str)
+    parser.add_argument("--port", nargs='?', help="Database port", type=str)
+    parser.add_argument("-f", "--file", help="Path to the csv file", type=str)
+    parser.add_argument(
+        "-c", "--collection",
+        help="Name of the collection to import data into. A new collection will be created if it does not exist.",
+        type=str
+    )
+    args = vars(parser.parse_args())
+
+    os.system(
+        "sudo docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' mysql > tmp"
+    )  # todo not the best solution but it works
+    host = open('tmp', 'r').read().strip()
+    os.remove('tmp')
+
+    if not host:
+        print("Could not automatically determine the database host IP address.")
+        print("Please provide host and port with the --host and --port arguments.")
+        exit()
+    else:
+        args["host"] = host
+        args["port"] = DEFAULT_PORT
```

Comment:
Please replace with default=DEFAULT_PORT, wonder if this const is required at all.